app/src/main/java/org/stepic/droid/web/ApiImpl.java

```diff
@@ -475,8 +477,8 @@ public void loginWithSocial(final FragmentActivity activity, ISocialType type) {
         String socialIdentifier = type.getIdentifier();
         String url = config.getBaseUrl() + "/accounts/" + socialIdentifier + "/login?next=/oauth2/authorize/?" + Uri.encode("client_id=" + config.getOAuthClientId(TokenType.social) + "&response_type=code");
         Uri uri = Uri.parse(url);
-        final Intent intent = new Intent(Intent.ACTION_VIEW).setData(uri);
-        activity.startActivity(intent);
+        final Intent intent = new Intent(activity, SocialAuthActivity.class).setData(uri);
+        activity.startActivityForResult(intent, LaunchActivity.REQUEST_CODE_SOCIAL_AUTH);
```

Comment:
activity should not know about request codes. Please, use `screenManager` here for separation of concerns